src/TestFramework/TestFramework/Assertions/CollectionAssert.cs

```diff
@@ -1652,5 +2701,23 @@ private class ObjectComparer : IComparer
     {
         int IComparer.Compare(object? x, object? y) => Equals(x, y) ? 0 : -1;
     }
+
+    private class ComparerWrapper<T> : IEqualityComparer<T>
+    {
+        private readonly IComparer _comparer;
+
+        public ComparerWrapper([NotNull] IComparer? comparer)
+        {
+            Assert.CheckParameterNotNull(comparer, "Assert", "comparer", string.Empty);
+            _comparer = comparer;
+        }
+
+        bool IEqualityComparer<T>.Equals(T? x, T? y) =>
+            // Use the comparer to check if the two items are equal
+            _comparer?.Compare(x, y) == 0;
+
+        public int GetHashCode(T obj) =>
```

Comment:
If I do recall well, the `GetHashCode` is sometimes called instead of `Equals` (e.g. key of `Dictionary`) so the logic should be updated to return the same hashcode in case of equality.

@MarcoRossignoli could you confirm or infirm?